### DISCUSSION TITLE
How to use WellectConnect in vanilla html/js?? plz

### DISCUSSION THREAD

**Discussion by Blockjinmoo** [WalletConnect/walletconnect-monorepo#1807] [Developer Support]
Hello, my project is based on vanilla html/js, server is Apache, not node.js.
How can I use WellectConnect?

I read this tutorial, but it seem that I need to make a server based on node.js and install the libs by npm/Yarn.
https://docs.walletconnect.com/2.0/web3modal/html-js/installation

Is there any method to use WellectConnect by just write <script> tag in html?

Thank you

**Comment by scorpionmit01** [comment]
There is not a method, but it can be done...  The documentation is a bit lacking for this, but it is doable.

You will need a Node.js Server with NPM to "build" your project, you can then deploy on plain HTML/JS.

Setup a Development Environment with Node.js, NPM, Typescript, and Webpack. You don't NEED to learn these tools in detail, but you'll need to follow someone's tutorial to build a hello world app with them, then build your wallet connect code.

Basically, you use Webpack to "compile" (npx webpack) it into a single file (you need to set Max Chunks to 1). Then you just load the JS file and it works as normal Javascript.

I'm NOT experienced with any of those environments and I learned them enough to get it done in about 8 hours, mostly because of poor documentation on the Internet, I think in the end I needed to edit like 8-12 lines of code and run a few commands.

But a normal HTML/JS solution? Where you have a Queue function that if the WalletConnect is setup first, it runs the queued commands, and if not loads them on load? The project is pretty far from it. It's probably, code-wise, pretty easy to implement, but doesn't seem a priority at this time.

**Reply by Blockjinmoo** [reply]
I installed NODEJS. And based on the https://docs.walletconnect.com/2.0/web3modal/html-js/installation site, we ran "npm install @web3modal/etherum @web3modal/html @wagmi/core ethers". However, this could not even be started due to an error. Do you know how to solve this error?

The error contents are as follows.

PS C:\Users\user\Desktop> npm install @web3modal/ethereum @web3modal/html @wagmi/core ethers
npm ERR! code ENOENT
npm ERR! syscall spawn git
npm ERR! path git
npm ERR! errno -4058
npm ERR! enoent An unknown git error occurred
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\user\AppData\Local\npm-cache\_logs\2022-12-28T23_43_34_624Z-debug-0.log

**Comment by scorpionmit01** [comment]
You need to install Webpack to compile it.Sent from my iPhoneOn Dec 28, 2022, at 6:46 PM, Blockjinmoo ***@***.***> wrote:﻿
I installed NODEJS. And based on the https://docs.walletconnect.com/2.0/web3modal/html-js/installation site, we ran "npm install @web3modal/etherum @web3modal/html @wagmi/core ethers". However, this could not even be started due to an error. Do you know how to solve this error?
The error contents are as follows.
PS C:\Users\user\Desktop> npm install @web3modal/ethereum @web3modal/html @wagmi/core ethers
npm ERR! code ENOENT
npm ERR! syscall spawn git
npm ERR! path git
npm ERR! errno -4058
npm ERR! enoent An unknown git error occurred
npm ERR! enoent This is related to npm not being able to find a file.
npm ERR! enoent
npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\user\AppData\Local\npm-cache_logs\2022-12-28T23_43_34_624Z-debug-0.log
